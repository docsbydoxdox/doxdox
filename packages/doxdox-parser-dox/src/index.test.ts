import { parseString } from './index';

describe('dox parser', () => {
    it('parse example jsdoc headers', async () => {
        await expect(
            parseString(
                'lib/utils/index.js',
                `/**
 * Finds file in path.
 *
 *     console.log(await findFileInPath('./', 'package.json'));
 *     console.log(await findFileInPath('../', 'package.json'));
 *     console.log(await findFileInPath('~/git/github/doxdox/', '.package.json'));
 *
 * @param {string} [input] Directory to check for file.
 * @param {string?} [fileName = 'package.json'] File name to check for.
 * @return {Promise<string | null>} Path to package.json file.
 * @public
 */

const findFileInPath = async (input, fileName = 'package.json') => {};

/**
 * Get the root directory of the package, supplied path or URL.
 *
 * @param {string?} [url] Optional path or URL.
 * @return {string} Directory path.
 * @public
 */

const getRootDirPath = (url) => {};`
            )
        ).resolves.toEqual(
            expect.objectContaining({
                path: 'lib/utils/index.js',
                methods: expect.arrayContaining([
                    expect.objectContaining({
                        type: 'declaration',
                        fullName: 'findFileInPath(input, fileName)',
                        name: 'findFileInPath',
                        params: expect.arrayContaining([
                            expect.objectContaining({
                                name: 'input',
                                types: ['string']
                            }),
                            expect.objectContaining({
                                name: 'fileName',
                                types: ['string']
                            })
                        ]),
                        private: false,
                        returns: expect.arrayContaining([
                            expect.objectContaining({
                                name: null,
                                description: 'Path to package.json file.',
                                types: ['Promise.<string|null>']
                            })
                        ]),
                        slug: 'lib-utils-index-js-findfileinpath'
                    }),
                    expect.objectContaining({
                        type: 'declaration',
                        fullName: 'getRootDirPath(url)',
                        name: 'getRootDirPath',
                        params: expect.arrayContaining([
                            expect.objectContaining({
                                name: 'url',
                                types: ['string']
                            })
                        ]),
                        private: false,
                        returns: expect.arrayContaining([
                            expect.objectContaining({
                                name: null,
                                description: 'Directory path.',
                                types: ['string']
                            })
                        ]),
                        slug: 'lib-utils-index-js-getrootdirpath'
                    })
                ])
            })
        );
    });
    it('parse empty string', async () => {
        await expect(parseString('test.js', '')).resolves.toEqual(
            expect.objectContaining({
                path: 'test.js',
                methods: []
            })
        );
    });
});
