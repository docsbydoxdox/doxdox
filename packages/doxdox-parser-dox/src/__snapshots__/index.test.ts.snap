// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dox parser parse empty file 1`] = `
{
  "methods": [],
  "path": "./test/mocks/empty.js",
}
`;

exports[`dox parser parse example jsdoc headers (declaration methods) 1`] = `
{
  "methods": [
    {
      "description": "Finds file in path.

    console.log(await findFileInPath('./', 'package.json'));
    console.log(await findFileInPath('../', 'package.json'));
    console.log(await findFileInPath('~/git/github/doxdox/', '.package.json'));",
      "fullName": "findFileInPath(input, fileName)",
      "name": "findFileInPath",
      "params": [
        {
          "description": "Directory to check for file.",
          "name": "input",
          "types": [
            "string",
          ],
        },
        {
          "description": "= 'package.json'] File name to check for.",
          "name": "fileName",
          "types": [
            "string",
          ],
        },
      ],
      "private": false,
      "returns": [
        {
          "description": "Path to package.json file.",
          "name": null,
          "types": [
            "Promise.<string|null>",
          ],
        },
      ],
      "slug": "test-mocks-declaration-js-findfileinpath",
      "type": "declaration",
    },
    {
      "description": "Get the current working directory.",
      "fullName": "getCurrentWorkingDirectory()",
      "name": "getCurrentWorkingDirectory",
      "params": [],
      "private": false,
      "returns": [
        {
          "description": "Directory path.",
          "name": null,
          "types": [
            "string",
          ],
        },
      ],
      "slug": "test-mocks-declaration-js-getcurrentworkingdirectory",
      "type": "declaration",
    },
    {
      "description": "Get the root directory of the package, supplied path or URL.",
      "fullName": "getRootDirPath(url)",
      "name": "getRootDirPath",
      "params": [
        {
          "description": "Optional path or URL.",
          "name": "url",
          "types": [
            "string",
          ],
        },
      ],
      "private": false,
      "returns": [
        {
          "description": "Directory path.",
          "name": null,
          "types": [
            "string",
          ],
        },
      ],
      "slug": "test-mocks-declaration-js-getrootdirpath",
      "type": "declaration",
    },
  ],
  "path": "./test/mocks/declaration.js",
}
`;

exports[`dox parser parse example jsdoc headers (function methods) 1`] = `
{
  "methods": [
    {
      "description": "Get the current working directory.",
      "fullName": "getCurrentWorkingDirectory()()",
      "name": "getCurrentWorkingDirectory()",
      "params": [],
      "private": false,
      "returns": [
        {
          "description": "Directory path.",
          "name": null,
          "types": [
            "string",
          ],
        },
      ],
      "slug": "test-mocks-function-js-getcurrentworkingdirectory",
      "type": "function",
    },
    {
      "description": "Get the root directory of the package, supplied path or URL.",
      "fullName": "getRootDirPath()(url)",
      "name": "getRootDirPath()",
      "params": [
        {
          "description": "Optional path or URL.",
          "name": "url",
          "types": [
            "string",
          ],
        },
      ],
      "private": false,
      "returns": [
        {
          "description": "Directory path.",
          "name": null,
          "types": [
            "string",
          ],
        },
      ],
      "slug": "test-mocks-function-js-getrootdirpath",
      "type": "function",
    },
  ],
  "path": "./test/mocks/function.js",
}
`;
